tools/rbac-gen/main.go

```diff
@@ -0,0 +1,104 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"io/ioutil"
+	"log"
+	"strings"
+	"sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/declarative/pkg/manifest"
+	"sigs.k8s.io/yaml"
+
+)
+
+type roleStruct struct {
```

Comment:
It might be easier to use the structs from k8s.io/api here, e.g. https://github.com/kubernetes/api/blob/49be0e3344fe443eb3d23105225ed2f1ab1e6cab/rbac/v1/types.go#L167